app/controllers/jobs_controller.rb

```diff
@@ -16,29 +25,55 @@ def create
       flash[:success] = "You created #{@job.title} at #{@company.name}"
       redirect_to company_job_path(@company, @job)
     else
+      @errors = @job.errors.full_messages
+      @categories = Category.all
+      @page_setup = PageSetup.new(:jobs_new)
       render :new
     end
   end
 
   def show
     @job = Job.find(params[:id])
+    @company = Company.find(params[:company_id])
+    @job_category = @job.category
+    @job_comments = @job.comments.newest_first
+    @comment_count = @job_comments.count
```

Comment:
i'd be okay with you accessing this directly in the view rather than setting it here 👍